x-pack/plugins/enterprise_search/server/lib/indices/fetch_indices.test.ts

```diff
@@ -463,11 +475,16 @@ describe('fetchIndices lib function', () => {
         '*',
         true,
         true,
-        'search-'
+        { alias_pattern: 'search-', index_pattern: '.ent-search-engine-documents' }
       );
 
-      expect(returnValue).toEqual(
-        expect.arrayContaining(Object.keys(mockMultiStatsResponse.indices).map(getIndexReturnValue))
+      // this specific alias should not be returned because...
```

Comment:
This and other deleted variations of this string should be added to the `not containing` block above. See my other comment.

I feel on the next iteration we should overhaul the mock data we have to make them easier to work with as you had to do a lot of sorting and reordering here.

Reply:
This was a half-baked comment that I meant to finish and never did 😅 

I actually don't know if `alias-.ent-search-engine-documents-12345` should or should not be returned from `fetchIndices`... but honestly, it doesn't seem to actually matter in the UI, which is almost worrisome. Definitely need to refactor these functions in a future iteration!

Reply:
It would be an alias to a hidden index, which shouldn't be shown. For the UI doesn't matter as it doesn't have the information at all. We need E2E tests to make sure this works properly at some point. Up until then, we can only be strict on unit tests for both sides.

Reply:
Next iteration we should work on the test data we create as well in that case.